packages/internal/bridge/sdk/src/tokenBridge.ts

```diff
@@ -979,12 +1015,86 @@ export class TokenBridge {
  */
   public async getFlowRateWithdrawTx(req:FlowRateWithdrawRequest): Promise<FlowRateWithdrawResponse> {
     // eslint-disable-next-line no-console
-    console.log('stubbed response with req', req);
+
+    if (!req.recipient) {
+      throw new BridgeError(
+        `invalid recipient ${req.recipient}`,
+        BridgeErrorType.INVALID_RECIPIENT,
+      );
+    }
+
+    if (req.index < 0) {
+      throw new BridgeError(
+        `invalid index ${req.index}`,
+        BridgeErrorType.INVALID_FLOWRATE_INDEX,
+      );
+    }
+
+    const rootBridge = await withBridgeError<ethers.Contract>(
+      async () => {
+        const contract = new ethers.Contract(
+          this.config.bridgeContracts.rootERC20BridgeFlowRate,
+          ROOT_ERC20_BRIDGE_FLOW_RATE,
+          this.config.rootProvider,
+        );
+        return contract;
+      },
+      BridgeErrorType.INTERNAL_ERROR,
+    );
+
+    const pending:Array<RootBridgePendingWithdrawal> = await rootBridge.getPendingWithdrawals(req.recipient, [0]);
+
+    if (pending.length === 0) {
+      throw new BridgeError(
+        `no pending withdrawals found for ${req.recipient}`,
+        BridgeErrorType.FLOW_RATE_ERROR,
+      );
+    }
+
+    if (!pending[req.index]) {
+      throw new BridgeError(
+        `invalid index ${req.index}`,
+        BridgeErrorType.FLOW_RATE_ERROR,
+      );
+    }
+
+    // @TODO query timeout from contract (SMR-2090)
+    const timeoutEnd = pending[req.index].timestamp.toNumber() + (60 * 60 * 24);
```

Comment:
> (60 * 60 * 24);

I would suggest not hardcoding this, given the value could actually change on the bridge. I suggest querying `withdrawalDelay()` on the root bridge instead (this can be done once and reused across this class)

Reply:
yeah, I've just added a TODO here, the next ticket is about fetching the flowRate config. So once I've coded the function that retrieves those settings from the contract, I will swap this hardcoded value out.